### ISSUE TITLE
Documentation: Missing library to install while building docs

### ISSUE THREAD

**Issue by motatoes** [pyinstaller/pyinstaller#5744]
## Description of the issue
There is a missing library in the docs: https://pyinstaller.readthedocs.io/en/stable/development/documentation.html

`sphinx_autodoc_typehints` should be included (as seen in stacktrace)

## Context information (for bug reports)

* Output of `pyinstaller --version`:  N/A
* Version of Python: N/A
* Platform: e.g GNU/Linux (distribution), Windows (language settings), OS X
* Did you also try this on another platform? Does it work there? No

### Stacktrace / full error message

```
make html
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v3.5.4

Extension error:
Could not import extension sphinx_autodoc_typehints (exception: No module named 'sphinx_autodoc_typehints')
make: *** [html] Error 2
```

Please also see <https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs>
for more about what would use to solve the issue.

**Comment by bwoodsend** [comment]
Oops, that's my fault. It should have gone in 419f349dad721a253b19d9c596e251818132d6ba. I'll fix that now.

**Comment by bwoodsend** [comment]
Fixed in 7758827986b04249699d14d2c42c7b92b4d41ee5. Not sure how long it takes for the change to propagate to the website.
